comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +34,70 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+			valuesReference = new ImportantValues();
+			void sizeRead() //reads the size of each file in a directory
+			{
+				string[] files = new string[0];
+				files = Directory.GetFiles(directory); //Assigns each file in the directory to an array
+				foreach (string file in files)
+				{
+					valuesReference.fileSize += new FileInfo(file).Length;
+
+					valuesReference.fileDirectories[0] = valuesReference.fileDirectories[0] + file;
+					valuesReference.allSizes[0] = valuesReference.allSizes[0] + valuesReference.fileSize;
```

Comment:
I think I can see what you're trying to do here -- maybe you wanted to make these `List`s rather than arrays, and then `Add` the values to them rather than using mathematical addition?